Makefile

```diff
@@ -31,19 +31,24 @@ SVSM_MEM	:= 0x10000000
 LDS_FLAGS	+= -DSVSM_GPA="$(SVSM_GPA)"
 LDS_FLAGS	+= -DSVSM_MEM="$(SVSM_MEM)"
 
-.PHONY: all doc prereq clean superclean
+EXT_LIBS := external/libcrt/libcrt.a
+
+.PHONY: all doc prereq clean clean_all superclean $(EXT_LIBS)
```

Comment:
@tlendacky Could you elaborate? 

If we want the Makefile to rebuild the external libraries whenever necessary, I think this may be our best option for now. By adding `$(EXT_LIBS)` to .PHONY we force their targets to always run, which forces their Makefile to check if the library needs to be rebuilt. The only downside I see is that `make` will always regenerate the SVSM binary. From a performance perspective, that should be fine. 

Most of the time, `make` will have nothing to do in the external libraries, but if we don't actively check that, a libcrt file update may get unnoticed.

Another way to accomplish that would be to move some of the logic from the libcrt Makefile to the SVSM Makefile, however we would not be able to do the same thing for the openssl Makefile because we don't own it.

Reply:
If you use something like libcrt as the .PHONY and add that to the all: target before the $(SVSM_BIN) that allows the rebuild of the external without necessarily always rebuilding svsm.bin, e.g.:

.PHONY: all doc prereq clean clean_all superclean libcrt

all: .prereq libcrt svsm.bin

svsm.bin: svsm.bin.elf
    ...
svsm.bin.elf: $(EXT_LIBS) $(OBJS) src/start/svsm.lds
   ...

libcrt:
    $(MAKE) ...

Then (at least for a default or 'all' make) you'll always go down the libcrt path, but only update svsm.bin if anything changed. I think that should work.

It's probably not a big deal right now and the build process can be updated later if we find a better way, so I'll leave it up to you.

Reply:
@tlendacky You are right. I ran some experiments and something like this seems to work the way we want. The SVSM binary is regenerated only if necessary. Additionally, changes to either libcrt files or the openssl Makefile triggers the library to be rebuilt.

Thank you, I will push these changes today later.

```
.PHONY: all doc prereq clean clean_all superclean libcrt libcrypto

all: .prereq libcrt libcrypto svsm.bin

external/libcrt/libcrt.a: libcrt

libcrt:
	$(MAKE) -C external/libcrt

external/openssl/libcrypto.a: libcrypto

libcrypto: external/openssl/Makefile libcrt
	$(MAKE) -C external/openssl -j$$(nproc)

svsm.bin.elf: $(EXT_LIBS) $(OBJS) src/start/svsm.lds
	$(GCC) $(LD_FLAGS) -o $@ $(OBJS) -Wl,-u,malloc -Wl,--start-group $(EXT_LIBS) -Wl,--end-group
```